### DISCUSSION TITLE
Linguist doesn't update language on new branch?

### DISCUSSION THREAD

**Discussion by timelessnesses** [github-linguist/linguist#6357] [Q&A] [answered]
So I tried rewriting my code a week ago and I rewrite it in rust.  
Normally linguist should update in few minutes (or even seconds) but the rewrite I pushed to new repository  
and that new branch didn't get updated by linguist for a week now (screenshot here)  
![image](https://user-images.githubusercontent.com/59832159/229420903-4fc9ecb4-d635-4bf8-9240-f989aeddcc43.png)  couldn't find a similar openning issues though  
[repository](https://github.com/timelessnesses/countable/tree/rust)

**Comment by lildude** [comment, marked as answer]
> Linguist doesn't update language on new branch?

No. Linguist only runs against the default branch, which in your case is `main`.

**Reply by timelessnesses** [reply]
Can I ask why GitHub doesn't run Linguist on different branch than main?

**Reply by lildude** [reply]
It can run on any branch. It will however only analyse and show the stats for the [default branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#about-the-default-branch), which can be any branch you want. So if you want to show the stats for the `rust` branch, change your default branch from `main` to `rust` and push another change to your repo.

**Reply by timelessnesses** [reply]
thanks for information!
